rpc-tests/rpc/zkVerify/submitProof/index.test.ts

```diff
@@ -0,0 +1,109 @@
+import 'dotenv/config';
+import { ApiPromise, WsProvider, SubmittableResult } from '@polkadot/api';
+import { Keyring } from '@polkadot/keyring';
+import { describe, test, expect, beforeAll, afterAll } from '@jest/globals';
+import { proofs } from '../proofs';
+import { createApi, handleEvents, waitForAttestationId, waitForNewAttestation, waitForNodeToSync } from '../utils';
+
+const requiredEnvVariables: string[] = ['WEBSOCKET', 'PRIVATE_KEY'];
+
+requiredEnvVariables.forEach((envVar) => {
+    if (!process.env[envVar]) {
+        throw new Error(`Required environment variable ${envVar} is not set.`);
+    }
+});
+
+describe('Proof Submission and Event Handling', () => {
+    let api: ApiPromise;
+    let provider: WsProvider;
+    let startTime: number;
+
+    beforeAll(async () => {
+        provider = new WsProvider(process.env.WEBSOCKET as string);
+        api = await createApi(provider);
+        await waitForNodeToSync(api);
+    });
+
+    afterAll(async () => {
+        if (provider) {
+            await provider.disconnect();
+        }
+        if (api) {
+            await api.disconnect();
+        }
+    });
+
+    Object.entries(proofs).forEach(([proofType, { pallet, validProof, invalidProof }]) => {
+        test(`should successfully accept a ${proofType} proof, emit a NewAttestation event`, async () => {
+            startTime = Date.now();
+            const keyring = new Keyring({ type: 'sr25519' });
+            const account = keyring.addFromUri(process.env.PRIVATE_KEY as string);
+
+            const submitProof = api.tx[pallet].submitProof(validProof);
+
+            let proof_leaf: string | null = null;
+            let attestation_id: string | null = null;
+
+            await new Promise<void>((resolve, reject) => {
+                submitProof.signAndSend(account, async ({ events, status, dispatchError }: SubmittableResult) => {
```

Comment:
It seams to you you didn't catch errors here: On my side I tried it with default evn and that tests don't be able to catch the error and reporting it (`///Alice` is not a valid account on testnet)